src/api/auth.rs

```diff
@@ -25,28 +22,23 @@ pub fn routes() -> Router<AppState> {
 }
 
 async fn auth_login(State(state): State<AppState>) -> Result<Redirect, RouteError> {
-    Ok(Redirect::permanent(&get_redirect_url(
-        &state.config.external.steam_realm,
-        &state.config.external.steam_return_path,
-    )?))
+    Ok(Redirect::permanent(state.steam_openid.get_redirect_url()))
 }
 
 async fn auth_return(
     State(state): State<AppState>,
-    Query(mut query): Query<VerifyForm>,
+    Query(query): Query<String>,
```

Comment:
this fails
for getting the query as a raw string, it should be [`RawQuery`](https://docs.rs/axum/latest/axum/extract/struct.RawQuery.html)